Checkpoint Syslog Filebeat module

P1llus opened the pull request:
## What does this PR do?

This PR adds checkpoint filebeat module.

The difference between this module and the CEF checkpoint module is that this is utilizing the syslog output format instead of CEF.

This output format supports a much larger set of fields from Checkpoint.

This is a collaboration work with @adriansr .

## Why is it important?

Adding more supported products to the filebeat portfolio.

## Checklist

<!-- Mandatory
Add a checklist of things that are required to be reviewed in order to have the PR approved

List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
-->

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

Closes #16041

Nosetests run successfully:
`INTEGRATION_TESTS=1 BEAT_STRICT_PERMS=false TESTING_FILEBEAT_MODULES=checkpoint nosetests -v -s tests/system/test_xpack_modules.py`

elasticmachine commented:
Pinging @elastic/siem (Team:SIEM)

leehinman commented on x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json:
```diff
@@ -0,0 +1,974 @@
+      }
+    },
+    {
+      "set": {
```
event.category should be an array.

P1llus replied:
Modified the line to an array

leehinman commented on x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json:
```diff
@@ -0,0 +1,974 @@
+      }
+    },
+    {
+      "set": {
```
event.category should be an array.

P1llus replied:
Modified the line to an array

leehinman commented on x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json:
```diff
@@ -0,0 +1,974 @@
+    },
+    {
+      "set" : {
+        "field": "event.outcome",
```
this should probably be "event.type = allowed" instead of event.outcome.  #16637 has a good discussion about event.type vs event.outcome.  Also event.type should be an array.

P1llus replied:
Changed to use event.outcome success still, since the action performed by the firewall was successful, but also added event.type as allowed and connection

leehinman commented on x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json:
```diff
@@ -0,0 +1,974 @@
+    },
+    {
+      "set" : {
+        "field": "event.outcome",
```
this should probably be event.type = denied, and even.type is an array

P1llus replied:
Changed to use event.outcome success, since the action performed by the firewall was successful, but also added event.type as denied and connection

leehinman commented on x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json:
```diff
@@ -0,0 +1,974 @@
+      "set" : {
+        "field": "event.kind",
+        "value": "alert",
+        "if": "['Prevent', 'Detect', 'Quarantine'].contains(ctx.checkpoint?.rule_action)"
```
can we use Prevent, Detect or Quarantine to determine if "malware" or "intrusion_detection" should be added to event.category?

P1llus replied:
Added malware if malware_action was not null + intrusion_detection if the Checkpoint.rule_action is prevent or detect (as those are the only used by IPS)

leehinman commented on x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json:
```diff
@@ -0,0 +1,974 @@
+    {
+      "rename": {
+        "field": "checkpoint.file_md5",
+        "target_field": "file.hash.md5",
```
if any of the hashes exist they should be added to "related.hash"

P1llus replied:
Done!

leehinman commented on x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json:
```diff
@@ -0,0 +1,974 @@
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.malware_action",
```
we could probably use the presense of "checkpoint.malware_action" to add "malware" to event.category before you rename it.

leehinman commented on x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json:
```diff
@@ -0,0 +1,974 @@
+    },
+    {
+      "set": {
+        "field": "source.ip",
```
all the ip addresses should also be copied to "related.ip"

P1llus commented:
All changes from comments has been applied, new expected.json generated, and nosetest has been run again.

andrewkroh commented on CHANGELOG.next.asciidoc:
```diff
@@ -254,6 +254,8 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Improve ECS categorization field mappings for mysql module. {issue}16172[16172] {pull}17491[17491]
 - Release Google Cloud module as GA. {pull}17511[17511]
+- Added new Checkpoint Syslog filebeat module. {pull}17682[17682]
+=======
```
```suggestion
```

andrewkroh commented on x-pack/filebeat/module/checkpoint/_meta/docs.asciidoc:
```diff
@@ -0,0 +1,158 @@
+== Checkpoint module
+beta[]
+
+This is a module for Checkpoint firewall logs. It supports logs from the Log Exporter in the Syslog format.
```
I think we should add a link to the CEF module and mention that it can support receiving data from CheckPoint in CEF format. If there's a trade-off that would be good to explain.

Likewise link from CEF to Checkpoint.

P1llus commented:
Added updated documentation changes as per @andrewkroh comments. Documentation from CEF to checkpoint module will be done in a separate PR.

tonymeehan commented on x-pack/filebeat/module/checkpoint/firewall/test/checkpoint.log-expected.json:
```diff
@@ -0,0 +1,5230 @@
+    {
+        "@timestamp": "2020-03-29T13:19:20Z",
+        "checkpoint.sys_message": ":\"The eth0 interface is not protected by the anti-spoofing feature. Your network may be at risk",
+        "event.category": [
```
👍

P1llus replied:
Just Anti-Spoofing being disabled, harder to setup a full WAN/LAN when you are running on a small virtual server 👍

P1llus replied:
Seems like the format for checkpoint.sys_message is different than the rest during errors and alerts, will add a fix for this.

tonymeehan replied:
I was commenting that it's good to see event.category as an array!

P1llus commented:
@tonymeehan 

I have added a workaround for this issue, as its not in the parser itself but rather a bug on the Checkpoint side.

I have updated the dataset to have at least 1 properly formatted and 1 incorrectly formatted message, to ensure the parser works for both. Re-ran nosetests afterwards.

tonymeehan commented:
@P1llus really awesome to see this PR!

hazcod commented:
@P1llus / @tonymeehan : This seems to not include any `ssl` configuration for input?
Any way to overwrite it?
